README.md

```diff
@@ -1,94 +1,235 @@
-
-### Basic Tests
-
-The project provides some very basic test skeletons that you can use as a starting point for building tests for your mixin.
-
-### Makefile
-
-The project also includes a Makefile that will can be used to both build and install the mixin. The Makefile also includes a TODO `publish` target that shows how you might publish the mixin and generate an mixin feed for easily sharing your mixin.
\ No newline at end of file
+# A Docker Mixin for Porter
+
+This is a Docker mixin for Porter. The mixin provides the Docker CLI. 
+
+## Mixin Declaration
+
+To use this mixin in a bundle, declare it like so:
+
+```yaml
+mixins:
+- docker
+```
+
+## Mixin Commands
+The commands available are docker pull, push, build, run, remove, and login. 
+
+## Mixin Syntax & Examples
+The same syntax applies for install, upgrade, and uninstall.
+
+### Docker pull
+
+#### Syntax
+You can specify either the tag or the digest.
+```yaml
+- docker:
+    description: "Description of the command"
+    pull:
+      name: IMAGE_NAME
+      tag: IMAGE_TAG
+      digest: IMAGE_DIGEST
+      arguments:
+      - arg1
+      - arg2
+      flags:
+        FLAGNAME: FLAGVALUE
+        REPEATED_FLAG:
+        - FLAGVALUE1
+        - FLAGVALUE2
+      outputs:
+        - name: NAME
+          hostname: HOSTNAME
+```
+
+#### Example
+````yaml
+- docker:
+    description: "Install Whalesay"
+    pull:
+      name: docker/whalesay
+      tag: latest
+````
+
+### Docker push
+
+#### Syntax
+```yaml
+- docker:
+    description: "Description of the command"
+    push:
+      name: IMAGE_NAME
+      tag: IMAGE_TAG
+      arguments:
+      - arg1
+      - arg2
+      flags:
+        FLAGNAME: FLAGVALUE
+        REPEATED_FLAG:
+        - FLAGVALUE1
+        - FLAGVALUE2
+      outputs:
+        - name: NAME
+          hostname: HOSTNAME
+```
+
+#### Example
+````yaml
+- docker:
+    description: "Push image"
+    push:
+      name: gmadhok/cookies
+      tag: v1.0
+````
+
+### Docker build
+
+#### Syntax
+```yaml
+- docker:
+    description: "Description of the command"
+    build:
+      tag: IMAGE_TAG
+      file: Dockerfile #OPTIONAL
+      path: PATH #defaults to "." OPTIONAL
+      arguments:
+      - arg1
+      - arg2
+      flags:
+        FLAGNAME: FLAGVALUE
+        REPEATED_FLAG:
+        - FLAGVALUE1
+        - FLAGVALUE2
+      outputs:
+        - name: NAME
+          hostname: HOSTNAME
+```
+
+#### Example
+````yaml
+- docker:
+    description: "Build image"
+    build:
+      tag: "gmadhok/cookies:v1.0"
+      file: Dockerfile
+````
+
+### Docker run
+
+#### Syntax
+```yaml
+- docker:
+    description: "Description of the command"
+    run:
+      image: IMAGE
+      name: NAME 
+      detach: BOOL #defaults to false
+      ports:
+        - host: HOST_PORT
+          container: CONTAINER_PORT
```

Comment:
Let's make it clear that they should type in a number here (instead of a string like 'http')

```suggestion
      ports:
        - host: NUMBER # porter exposed on the host
          container: NUMBER # port exposed by the container
```